### ISSUE TITLE
Invite Brave Search users to participate in the Web Discovery Project 

### ISSUE THREAD

**Issue by rebron** [brave/brave-browser#18488]
## Description 

Provide a way to offer Brave Search users to participate in the Web Discovery Project. For users who have Brave Search as their default search engine and visit https://search.brave.com, on the nth and subsequent nth visit display a notification prompt to let them know about Web Discovery Project and an offer to participate to help make Brave Search better. 

## Design
<img width="420" alt="Screen Shot 2021-10-04 at 10 41 43 AM" src="https://user-images.githubusercontent.com/2388823/135898486-458ea577-9f73-49aa-a2d9-0c3908d2dc01.png">

Turn on toggles brave://settings/search Web Discovery Project setting to on.
No Thanks closes dialog.

Learn more opens new tab to: 
`https://www.brave.com/browser/privacy/#web-discovery-project`

See Figma -> https://www.figma.com/file/iuKMDbanhcrO1VSBsssZbb/Desktop-Brave-Search?node-id=36%3A325

## Reproduces how often: 

n/a

## Brave version (brave://version info)

n/a

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a
- Can you reproduce this issue with the beta channel? n/a
- Can you reproduce this issue with the nightly channel? n/a

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? n/a
- Does the issue resolve itself when disabling Brave Rewards? n/a
- Is the issue reproducible on the latest version of Chrome? n/a

**Comment by rmcfadden3** [comment]
@rebron — looks good. Only discrepancy I see is we removed the "user-driven" thing, since we got some feedback (forget from where) that that phrase was kinda meaningless. So final copy =

H1: Want to make Brave Search even better?
Body: Opt in to the Web Discovery Project to anonymously contribute some search & browsing data, and improve Brave Search results for all. Privacy-preserving. And zero effort. Learn more
CTA: Turn on | No thanks | Don’t ask again

And that "Learn more" points to https://brave.com/privacy/browser/#web-discovery-project
